lib/spack/spack/test/cmd_extensions.py

```diff
@@ -97,6 +108,40 @@ def hello_folks():
         del sys.modules[module_name]
 
 
+@pytest.fixture()
+def subcommand_root(tmpdir):
+    root = tmpdir.mkdir('subcommand_root')
+    return root
+
+
+def _ensure_package(base, *parts):
+    bits = []
+    for bit in parts[:-1]:
+        bits.append(bit)
+        base.ensure(*(bits + ['__init__.py']))
+    return base.ensure(*parts)
+
+
+@pytest.fixture()
+def hello_world_sub_no_cmd(subcommand_root):
+    """Failed attempt at a subcommand (bad location)."""
+    hello = _ensure_package(subcommand_root, 'bad', 'hello.py')
+    hello.write(_hello_world_cmd_text)
+    sys.path.insert(0, str(subcommand_root))
+    yield
+    sys.path.pop()
```

Comment:
@chissg Not sure how I missed this before but this is causing all the import failures. If `sys.path` is a list, then `pop` will remove the last entry (i.e. some random module).

Even if this data structure popped from the front, if any module gets imported during the yield, the wrong entry would be removed. 

At minimum, this will have to use `sys.path.remove(str(subcommand_root))`.

Reply:
D'oh! Changed as suggested. Tests running now.